src/components/App/index.js

```diff
@@ -8,19 +8,47 @@ import JoinGame from '../JoinGame';
 import Lobby from '../Lobby';
 import Main from '../Main';
 import ErrorScreen from '../ErrorScreen';
+import Cable from 'actioncable';
+
 
 export class App extends Component {
   constructor(){
     super()
     this.state={
-      socket: "0000"
+      token: "000"
     }
   }
 
-  handleWebSocket = ({name,value}) => {
+  handleUserInit = ({name,token}) => {
+    console.log(token)
    this.setState({
-     socket:value
-   })
+     token
+   }, () => this.createCable(token) )
+   
+  }
+  createCable = (token) => {
+    if(this.state.token === token ){
+      let cable = Cable.createConsumer(`ws://localhost:3000/cable/${token}`);
+      console.log(cable)
+      this.chats = cable.subscriptions.create({
+        channel: 'GameDataChannel'
+      }, {
+        connected: () => {console.log("connected")},
+        disconnected: () => { console.log("disconnected")},
+        rejected: () => { console.log("rejected")},
+        received: (data) => {
+          console.log(data)
+          // let chatLogs = this.state.chatLogs;
+          // chatLogs.push(data);
+          // this.setState({ chatLogs });
+        },
+        create: function(chatContent) {
+          this.perform('create', {
+            content: "hello"
+          });
```

Comment:
It looks like a lot of this is trial and error. But when you get the chance could you run me through your thought process on tackling this. And what you found works best?

Reply:
This isnt trial and error so much as we needed to see what message we were getting back when trying to connect, on the backend. We had an issue with the socket immediately disconnecting each time and solved it with this.